pkg/agent/handler/accessrequest.go

```diff
@@ -60,10 +60,18 @@ func (h *accessRequestHandler) Handle(action proto.Event_Type, meta *proto.Event
 		return nil
 	}
 
-	if ok := shouldProcess(ar.Status.Level, ar.Metadata.State); !ok {
-		return nil
+	if ok := shouldProcessPending(ar.Status.Level, ar.Metadata.State); ok {
+		return h.onPending(ar)
+	}
+
+	if ok := shouldProcessDeleting(ar.Status.Level, ar.Metadata.State, len(ar.Finalizers)); ok {
```

Comment:
When a resource is in deleting state we end up creating another update event to remove the finalizer. So, I added a check to see if there are finalizers before deprovisioning.